components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManagerTest.java

```diff
@@ -0,0 +1,204 @@
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.util.HashMap;
+
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.MockitoAnnotations.initMocks;
+import static org.powermock.api.mockito.PowerMockito.*;
+
+/*
+ * Unit tests for SCIMUserManager
+ */
+
+@PrepareForTest({IdentityDatabaseUtil.class, Connection.class, SCIMGroupHandler.class, IdentityUtil.class})
+@PowerMockIgnore("java.sql.*")
+public class SCIMUserManagerTest extends PowerMockTestCase {
+
+    @Mock
+    private SCIMGroupHandler mockedGroupHandler;
+
+    @Mock
+    private PreparedStatement mockedPreparedStatement;
+
+    @Mock
+    private UserStoreManager mockedUserStoreManager;
+
+    @Mock
+    private ClaimManager mockedClaimManager;
+
+    @Mock
+    private GroupDAO mockedGroupDAO;
+
+    @Mock
+    private Connection connection;
+
+    @Mock
+    private ResultSet mockedResultSet;
+
+    @Mock
+    private IdentityUtil mockedIdentityUtil;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+
+       initMocks(this);
+    }
+
+    @AfterMethod
+    public void tearDown() throws Exception {
+    }
+
+    @Test
+    public void testCreateUser() throws Exception {
+
+    }
+
+    @Test
+    public void testGetUser() throws Exception {
+    }
+
+    @Test
+    public void testDeleteUser() throws Exception {
+    }
+
+    @Test
+    public void testListUsersWithGET() throws Exception {
+    }
+
+    @Test
+    public void testListUsersWithPost() throws Exception {
+    }
+
+    @Test
+    public void testUpdateUser() throws Exception {
+    }
+
+    @Test
+    public void testGetMe() throws Exception {
+    }
+
+    @Test
+    public void testCreateMe() throws Exception {
+    }
+
+    @Test
+    public void testDeleteMe() throws Exception {
+    }
+
+    @Test
+    public void testUpdateMe() throws Exception {
+    }
+
+    @Test
+    public void testCreateGroup() throws Exception {
+    }
+
+    @Test(dataProvider = "groupName")
+    public void testGetGroup(Object roleName, String userStoreDomain, String expected) throws Exception {
+
+        whenNew(GroupDAO.class).withAnyArguments().thenReturn(mockedGroupDAO);
+        when(mockedGroupDAO.getGroupNameById(anyInt(), anyString())).thenReturn((String) roleName);
+
+        mockStatic(IdentityUtil.class);
+        when(IdentityUtil.extractDomainFromName(anyString())).thenReturn(userStoreDomain);
+
+        SCIMUserManager scimUserManager = new SCIMUserManager(mockedUserStoreManager, mockedClaimManager);
+        Assert.assertEquals(scimUserManager.getGroup("1234567", new HashMap<String, Boolean>()), expected);
+
+    }
+
+    @DataProvider(name = "groupName")
+    public Object[][] groupName() {
+        return new Object[][]{
+                {null, "userStoreDomain",null },
```

Comment:
Are you sure you want to set roleName as null?

If you only have one set of values to test, then having a dataprovider is an overkill IMO.